File: spec/data/complicated/config/software/curl.rb
#
# Copyright:: Copyright (c) 2012-2014 Chef Software, Inc.
# License:: Apache License, Version 2.0
#
# Licensed under the Apache License, Version 2.0 (the "License");
# you may not use this file except in compliance with the License.
# You may obtain a copy of the License at
#
#     http://www.apache.org/licenses/LICENSE-2.0
#
# Unless required by applicable law or agreed to in writing, software
# distributed under the License is distributed on an "AS IS" BASIS,
# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
# See the License for the specific language governing permissions and
# limitations under the License.
#

name "curl"
default_version "7.36.0"

dependency "zlib"
dependency "openssl"

source :url => "http://curl.haxx.se/download/curl-#{version}.tar.gz",
       :md5 => "643a7030b27449e76413d501d4b8eb57"

relative_path "curl-#{version}"

build do
  command ["./configure",
           "--prefix=#{install_dir}/embedded",
           "--disable-debug",
           "--enable-optimize",
           "--disable-ldap",
           "--disable-ldaps",
           "--disable-rtsp",
           "--enable-proxy",
           "--disable-dependency-tracking",
           "--enable-ipv6",
           "--without-libidn",
           "--without-gnutls",
           "--without-librtmp",
           "--with-ssl=#{install_dir}/embedded",
           "--with-zlib=#{install_dir}/embedded"].join(" ")

  command "make -j #{max_build_jobs}", :env => {"LD_RUN_PATH" => "#{install_dir}/embedded/lib"}
  command "make install"
end

File: config/software/curl.clj
;;
;; Author:: Adam Jacob (<adam@opscode.com>)
;; Author:: Christopher Brown (<cb@opscode.com>)
;; Copyright:: Copyright (c) 2010 Opscode, Inc.
;; License:: Apache License, Version 2.0
;;
;; Licensed under the Apache License, Version 2.0 (the "License");
;; you may not use this file except in compliance with the License.
;; You may obtain a copy of the License at
;; 
;;     http://www.apache.org/licenses/LICENSE-2.0
;; 
;; Unless required by applicable law or agreed to in writing, software
;; distributed under the License is distributed on an "AS IS" BASIS,
;; WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
;; See the License for the specific language governing permissions and
;; limitations under the License.
;;

(software "curl" :source "curl-7.21.2"
          :steps [
                  {:command "./configure"
                   :args ["--prefix=/opt/opscode/embedded"
                          "--disable-debug"
                          "--enable-optimize"
                          "--disable-ldap"
                          "--disable-ldaps"
                          "--disable-rtsp"
                          "--enable-proxy"
                          "--disable-dependency-tracking"
                          "--enable-ipv6"
                          "--without-libidn"
                          "--with-ssl=/opt/opscode/embedded"
                          "--with-zlib=/opt/opscode/embedded"]}
                  {:command "make" :env { "LD_RUN_PATH" "/opt/opscode/embedded/lib" }}
                  {:command "make" :args ["install"]}
                 ; {:command "ln"
                 ;  :args ["-sfn"
                 ;         "/opt/opscode/embedded/bin/curl-config"
                 ;         "/usr/local/bin/curl-config"]}
                  ])


